operations/configuration_management/Management_Node_Personalization.md

```diff
@@ -0,0 +1,143 @@
+configuration.
+
+The script currently requires that we provide it with the name of the CFS configuration applied
+to the management nodes. If all management nodes do not use the same CFS configuration, it is
+best to use the
+[Manual procedure to re-run node personalization on management nodes](#manual-procedure-to-re-run-node-personalization-on-management-nodes).
+The first step of this procedure checks whether all management nodes use the same CFS configuration.
+
+1. (`ncn-mw#`) Get the CFS configuration applied to all management NCNs.
+
+    ```bash
+    CFS_CONFIG_NAME="$(cray cfs components list --format json \
+        --ids "$(cray hsm state components list --role=management --format json \
+                 | jq -r '.Components | map(.ID) | join(",")')" \
+        | jq -r 'map(.desiredConfig) | unique | first')"
+    echo "${CFS_CONFIG_NAME}" | xargs -n 1
+    ```
+
+    If there is a single configuration applied to all management nodes, the output should look like this:
+
+    ```text
+    management-23.4.0
+    ```
+
+    If there is more than one CFS configuration listed, use the
+    [Manual procedure to re-run node personalization on management nodes](#manual-procedure-to-re-run-node-personalization-on-management-nodes)
+    instead of the scripted procedure.
+
+1. (`ncn-mw#`) Execute the script with the `CFS_CONFIG_NAME` variable set in the previous step.
+
+    ```bash
+    /usr/share/doc/csm/scripts/operations/configuration/apply_csm_configuration.sh \
+        --no-config-change --config-name "${CFS_CONFIG_NAME}" --clear-state
+    ```
+
+    Successful output will end with a message similar to the following:
+
+    ```text
+    Configuration complete. 9 component(s) completed successfully.  0 component(s) failed.
+    ```
+
+    The number reported should match the number of management NCNs in the system. If there are failures, see
+    [Troubleshoot CFS Issues](Troubleshoot_CFS_Issues.md).
+
+### Manual procedure to re-run node personalization on management nodes
+
+This procedure manually clears the CFS configuration state and error count across all the management
+node components at once. This causes CFS to reconfigure these components.
+
+1. (`ncn-mw#`) To re-run NCN personalization on all management NCNs at once, use the following loop:
+
+    ```bash
+    FAILED="" ; COUNT=0 ; \
+    for xname in $(cray hsm state components list \
+                        --role Management --type node \
+                        --format json |
+                jq -r .Components[].ID)
+    do
+        echo "Clearing CFS state of ${xname}"
+        cray cfs components update --error-count 0 --state '[]' --format json "${xname}" && let COUNT+=1 || FAILED+=" ${xname}"
+    done ; \
+    echo "Cleared CFS state on ${COUNT} nodes" ; \
+    [[ -z ${FAILED} ]] && echo "No errors" || echo "ERROR: There were errors clearing the CFS state for the following nodes:${xname}"
```

Comment:
```suggestion
    [[ -z ${FAILED} ]] && echo "No errors" || echo "ERROR: There were errors clearing the CFS state for the following nodes:${FAILED}"
```

Reply:
Good catch! I was just copying this as-is from the deleted `Perform_NCN_Personalization.md` file, but this does clearly look like a bug.